### DISCUSSION TITLE
Question about BreakMeshByBlock Works with Distributed Mesh

### DISCUSSION THREAD

**Discussion by chunhuizhao478** [idaholab/moose#24189] [Q&A Modules: Solid mechanics] [answered]
Hi all, This is a general question about CZM (breakmeshbyblock) works with distributed mesh. We currently have issues with extending our 2D CZM model into 3D due to memory constraints (limit in duplicated mesh)

I have found the following GitHub Discussion Issue page:  https://github.com/idaholab/moose/issues/12033

There is a discussion between Cody and Andrea about Pre-split mesh option to achieve distributed mesh functionality. I find it quite interesting and did a simple test in local. It seems like the interface information is saved correctly but no variables is updated at each save (as exodus file).

I wonder if this option is still in progress or it is due to my misunderstanding about the process. Thanks!

**Comment by lindsayad** [comment, marked as answer]
We do not currently support CZM with distributed mesh. However, https://github.com/idaholab/moose/issues/24140 aims to enable that capability

**Reply by chunhuizhao478** [reply]
I saw some cool developments going on in #24140... Just curious about the current progress on this

**Reply by lindsayad** [reply]
I'm not sure. Maybe @ttruster can weigh in

**Reply by ttruster** [reply]
Unfortunately I have not had free time to work on this and probably won't get to start until December 2023 at the earliest. It is still one of my goals; currently (back in May 2023) the implementation runs, but now error cases need to be handled.

I am also open to mentoring someone else on wrapping this up if there is an interest/need to have it sooner.
